### DISCUSSION TITLE
Couldn't build again after installing all the packages for the essay.

### DISCUSSION THREAD

**Discussion by erodios** [gstavrinos/ai_msc_robotic_systems_assignments#2] [Q&A] [answered]
![Screenshot from 2023-03-24 17-08-07](https://user-images.githubusercontent.com/56002896/227565213-bcdc47bd-c00d-45b9-9882-235143244c20.png)

**Comment by gstavrinos** [comment]
Αν τρέξεις την εντολή του rosdep, διορθώνεται το πρόβλημα;

**Comment by erodios** [comment, marked as answer]
Gia ayto to error ebala auto sto terminal "sudo apt-get install ros-humble-xacro" kai meta mou ebgale ena paromoio error gia to table tennis kai ebala auto sto terminal "sudo apt-get install ros-humble-odom-to-tf-ros2" kai auto "sudo apt-get install ros-humble-play-motion2-msgs".

Telos mou ebgale auto to problhma
CMake Error at ament_cmake_symlink_install/ament_cmake_symlink_install.cmake:100 (message):
ament_cmake_symlink_install_directory() can't find
'/home/erodios/ros2_ws/src/gazebo_table_tennis/worlds'
Call Stack (most recent call first):
ament_cmake_symlink_install/ament_cmake_symlink_install.cmake:362 (ament_cmake_symlink_install_directory)
cmake_install.cmake:46 (include).

To opoio elysa dhmiougontas ena folder "worlds" mesa sto gazebo_table_tennis.

**Reply by gstavrinos** [reply]
Για όλα τα errors που η λύση ήταν να κάνεις χειροκίνητα `apt-get install`, θα μπορούσες επίσης να τρέξεις `rosdep`, που βρίσκει αυτόματα όσα λέιπουν. Για το worlds, είναι δικό μου το λάθος, και θα κάνω commit σε λίγο, που το διορθώνει. Thanks!

**Comment by vvitzilaios** [comment]
I get this error even after running the rosdep command:

```
parallels@ubuntu-linux-22-04-desktop:/media/psf/Home/Documents/ros2_ws$ colcon build --symlink-install
Starting >>> diff_drive_controller
Starting >>> launch_pal
Starting >>> pal_gripper_controller_configuration
Starting >>> hey5_description
--- stderr: diff_drive_controller                                          
CMake Error at CMakeLists.txt:29 (find_package):
  By not providing "Findbackward_ros.cmake" in CMAKE_MODULE_PATH this project
  has asked CMake to find a package configuration file provided by
  "backward_ros", but CMake did not find one.

  Could not find a package configuration file provided by "backward_ros" with
  any of the following names:

    backward_rosConfig.cmake
    backward_ros-config.cmake

  Add the installation prefix of "backward_ros" to CMAKE_PREFIX_PATH or set
  "backward_ros_DIR" to a directory containing one of the above files.  If
  "backward_ros" provides a separate development package or SDK, be sure it
  has been installed.


---
Failed   <<< diff_drive_controller [0.35s, exited with code 1]
Aborted  <<< hey5_description [0.38s]
Aborted  <<< pal_gripper_controller_configuration [0.38s]
Aborted  <<< launch_pal [0.63s]                      

Summary: 0 packages finished [1.16s]
  1 package failed: diff_drive_controller
  3 packages aborted: hey5_description launch_pal pal_gripper_controller_configuration
  1 package had stderr output: diff_drive_controller
  25 packages not processed
```


<img width="1255" alt="image" src="https://github.com/gstavrinos/ai_msc_robotic_systems_assignments/assets/118114473/553cf8ff-b3e1-4271-8731-8365de867269">

**Comment by gstavrinos** [comment]
Your workspace is totally messed up. You need to have all packages cloned in the src folder under the root of your workspace i.e. `ros2_ws/src/`. It appears that you have packages (and an src folder!) inside your install folder.

I would suggest completely removing the `install` and `build` folders, cloning this (assignments) repo, and running the dependencies script again. If everything goes as expected, you should have all packages cloned in the `src` folder and then you can continue the typical process of `rosdep` and `colcon`.

Keep me updated on the matter.

**Reply by vvitzilaios** [reply]
This was not the case. I have the packages inside the src folder which is under ros2_ws. I executed the commands as per the few instructions I found inside our lecture presentations. The Install folder just holds a ros2 installation. Anyway, I am going to repeat the same process you mentioned above even though that was the exact steps I did before. Hopefully I missed something.

**Reply by vvitzilaios** [reply]
Well, still getting the same error. :D

**Reply by gstavrinos** [reply]
Give me the output of the rosdep command

**Reply by vvitzilaios** [reply]
I managed to build. The error emerged as a result of the vm and how it recognises the shared folders. Apparently during dependency installation it could not resolve the path of the shared folder (f@ck me, right?) so I moved all the workspace to a not shared folder.

**Reply by gstavrinos** [reply]
Pretty weird, glad it worked though.

**Comment by vvitzilaios** [comment]
It is kinda unclear. Should we "install" the ros on the default linux path `/opt/ros/` 
or do we move the files inside our workspace install folder? 
right now I have installed inside default location but what if I want to run `ros2 launch tt_umpire sim_tt.launch.py` like the presentations describe?
